go/src/koding/kites/config/konfig.go

```diff
@@ -107,10 +114,31 @@ func (k *Konfig) Valid() error {
 }
 
 func (k *Konfig) ID() string {
-	hash := sha1.Sum([]byte(k.Endpoints.Koding.Public.String()))
+	hash := sha1.Sum([]byte(k.KodingPublic().String()))
```

Comment:
The panic is a correct behavior, it would mean invalid config/config.json.go generated thus a programmer error.